httpx/decoders.py

```diff
@@ -99,20 +113,30 @@ def __init__(self) -> None:
         ), "The 'brotlipy' or 'brotli' library must be installed to use 'BrotliDecoder'"
         self.decompressor = brotli.Decompressor()
 
-    def decode(self, data: bytes) -> bytes:
-        try:
-            if hasattr(self.decompressor, "decompress"):
-                return self.decompressor.decompress(data)
-            return self.decompressor.process(data)  # pragma: nocover
-        except brotli.error as exc:
-            raise DecodingError from exc
+    def decode(self, data: bytes, with_flush: bool = False) -> bytes:
```

Comment:
I'd strongly prefer not indenting everything in the body here, but instead just adding an early return in the empty case...

```python
def decode(self, data: bytes) -> bytes:
    if not data:
        return b""
... # Rest of the method unchanged.
```